votrfront/js/sorting.js

```diff
@@ -73,3 +74,184 @@ export function sortTable(items, columns, query, queryKey) {
 
   return [items, header];
 };
+
+export function sortTableCopyForObjectCol(items, columns, query, queryKey) {
+  var orderString = query[queryKey] || columns.defaultOrder;
+  var order = orderString ? orderString.split(/(?=[ad])/) : [];
+  var directions = order.map((o) => o.substring(0, 1) == 'a' ? 'asc' : 'desc');
+  var iteratees = order.map((o) => {
+    var {label, prop, process, preferDesc} = columns[o.substring(1)];
+    return (item) => (process || _.identity)(prop ? item[prop] : item);
+  });
+
+  items = _.orderBy(items, iteratees, directions);
+
+  function handleClick(event) {
+    var index = event.currentTarget.getAttribute('data-index');
+    var {label, prop, process, preferDesc} = columns[index];
+
+    var newOrder = _.without(order, 'a' + index, 'd' + index);
+    newOrder.unshift((
+      order[0] == 'a' + index ? 'd' :
+      order[0] == 'd' + index ? 'a' :
+      preferDesc ? 'd' : 'a') + index);
+
+    navigate({ ...query, [queryKey]: newOrder.join('') });
+  }
+
+  var header = <tr>
+    {columns.map(({label, prop, process, preferDesc, hiddenClass = []}, index) =>
+      <th key={index} data-index={index} onClick={handleClick}
+          className={`${hiddenClass.join(" ")} ` + 'sort ' + (order[0] == 'a' + index ? 'asc' :
+                                order[0] == 'd' + index ? 'desc' : '')}>
+        {label}
+      </th>
+    )}
+  </tr>;
+
+  return [items, header];
+};
+
+function toggleInfo(index) {
```

Comment:
[Veľké]
Myslím že tu nie je správne použiť LocalSettings. Takto by úplne každá SortableTable tabuľka zdieľala, ktoré riadky sú otvorené a ktoré nie. Ak máš otvorenú napr. prvú, tretiu a siedmu skúšku a prepneš sa na iný akademický rok, stále budú otvorené prvá, tretia a siedma (aj keď už sú úplne iné). Alebo časom keď aj iné sekcie prejdú na SortableTable, aj na ne by sa prenášalo, čo je práve otvorené.
Preto by to radšej mal byť lokálny stav komponentu (`this.state` a `this.setState`). Viď dokumentáciu Reactu a inšpiruj sa inými Votr komponentmi čo majú stav.

[Veľké a nepovinné]
Hmm, ale keď nie na toto, kde sme to chceli tie LocalSettings použiť? Aha, už si spomínam. Aspoň moja predstava bola, aby sa aj na mobile dať zapnúť mód že sa ukazujú všetky stĺpce, ako je to dnes. A to by mohlo byť v LocalSettings, lebo je to skôr globálne nastavenie pre užívateľa ako pre konkrétnu tabuľku.
(Prioritizácia: presunúť openedRows z LocalSettings do state treba, ale tento prepínač či vidno všetko je menej dôležitý a prípadne sa dá odložiť - radšej to chcem mať bez neho ako vôbec.)

Reply:
Pri tomto mi napadla otázka: keď zmením poradie riadkov, chcem aby sa opäť všetky riadky zatvorili, alebo nechám otvorené tie, čo boli predtým? (Teda ak bol otvorený prvý, tak ostane otvorený, aj keď v skutočnosti zobrazuje už iný predmet.)